code/parachain/frame/dex-router/src/lib.rs

```diff
@@ -483,24 +484,25 @@ pub mod pallet {
 				// other than `in_asset_itr.asset_id` gives us the out_asset_id
 				let out_asset_id = assets
 					.keys()
-					.copied()
-					.find(|a| *a != in_asset_itr.asset_id)
+					.find(|a| a != &&in_asset_itr.asset_id)
 					.ok_or(Error::<T>::NoRouteFound)?;
 				swap_result = T::Pablo::do_swap(
 					who,
 					*pool_id,
 					in_asset_itr,
-					AssetAmount::new(out_asset_id, T::Balance::zero()),
+					AssetAmount::new(*out_asset_id, T::Balance::zero()), /* blas: why balance::zero()? */
```

Comment:
this is the min receive amount, for dex router to work we need to allow any amount as the constraints are unknown for intermediate swaps as per current impl.